packages/ui/src/hooks/useObservable.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useState } from 'react'
+import { Observable } from 'rxjs'
+
+export function useObservable<T>(observable: Observable<T> | undefined, deps: readonly any[]) {
+  const [data, setData] = useState<T | undefined>(undefined)
+
+  useEffect(() => {
+    const subscription = observable?.subscribe(setData)
+    return () => subscription && subscription.unsubscribe()
+  }, deps)
+
+  return [data]
```

Comment:
Why an array?

Reply:
Hmmm, I've copied that part from the other project, but looks like I didn't give a second look at that. I don't see a reason actually.